gbdk-lib/examples/cross-platform/text_advanced_dialogue/src/main.c

```diff
@@ -0,0 +1,374 @@
+
+                // Wait a little bit
+                vsyncMultiple(15);
+
+                // Clear the previous line of text
+                fill_winbkg_rect(1,DIALOGUE_BOX_Y+2,INNER_DIALOGUE_BOX_WIDTH,1,0);
+
+                // Draw on the first row of our dialogue box
+                set_text_tiles(1,DIALOGUE_BOX_Y+1,INNER_DIALOGUE_BOX_WIDTH,1,copyBuffer);
+
+                copyBufferCount=0;
+
+                // Clear the copy buffer
+                for(uint8_t k=0;k<INNER_DIALOGUE_BOX_WIDTH;k++)copyBuffer[k]=0;
+
+                rowCount=2;
+
+            }
+            
+            // reset for the next row
+            vramAddress = get_winbkg_xy_addr(column,row+rowCount);
+            
+            columnSize=0;
+
+            // If we just started a new line, skip spaces
+            while(text[index]==' '){
+                index++;
+            }
+        }
+
+        #if !defined(SEGA) && !defined(NINTENDO_NES)
+
+            // Play a basic sound effect
+            NR10_REG = 0x34;
+            NR11_REG = 0x81;
+            NR12_REG = 0x41;
+            NR13_REG = 0x7F;
+            NR14_REG = 0x86;
+
+        #endif
+
+        vsyncMultiple(3);
+    }
+
+    HideDialgoueBox();
+    
+}
+
+
+
+void ClearScreen(void){
+    #if !defined(SEGA) && !defined(NINTENDO_NES)
+    // Game Gear doesn't have a window.
+    fill_win_rect(0,0,DEVICE_SCREEN_WIDTH,DEVICE_SCREEN_HEIGHT,0);
+    #endif
+    fill_bkg_rect(0,0,DEVICE_SCREEN_WIDTH,DEVICE_SCREEN_HEIGHT,0);
+}
+
+void main(void)
+{
+    DISPLAY_ON;
+    SHOW_BKG;
+
+    #if !defined(SEGA) && !defined(NINTENDO_NES)
+    SHOW_WIN;
+    #endif
+
+    fontTilesStart = DialogueBox_TILE_COUNT+1;
+    uint8_t emptyTile[TILE_SIZE_BYTES];
+    for(uint8_t i=0;i<TILE_SIZE_BYTES;i++)emptyTile[i]=0;
+
+    set_native_tile_data(0,1,emptyTile);
+
+    ClearScreen();
+
+    set_native_tile_data(1,DialogueBox_TILE_COUNT,DialogueBox_tiles);
+
+    
+    #if !defined(SEGA) && !defined(NINTENDO_NES)
+
+        NR52_REG = 0x80; // Master sound on
+        NR50_REG = 0xFF; // Maximum volume for left/right speakers. 
+        NR51_REG = 0xFF; // Turn on sound fully
+
+        // Completely hide the window
+        windowYPosition = (DEVICE_SCREEN_HEIGHT << 3)<<3;
+        MoveWindow();
+
+    #endif
+    
+    while(TRUE){
+
+        // We'll pass in one long string, but the game will present to the player multiple pages.
+        // By passing 3 as the final argument, the game boy will wait 3 frames between each character
+        DrawTextAdvanced("This is an how you draw text on the screen in GBDK. The code will automatically jump to a new line, when it cannot fully draw a word.  When you reach three lines, it will wait until you press A. After that, it will start a new page and continue. The code will also automatically start a new page after periods. For every page, the code will dynamically populate VRAM. Only letters and characters used, will be loaded into VRAM.");
```

Comment:
It would be helpful to put the "Press A button" info at the very beginning so it's clear that they should do that to continue displaying more text.